MovingBlocks/Terasology
# feat(player-config): migrate `PlayerConfig` to AutoConfig

### Contains

1. Migration `PlayerConfig` to `AutoConfig`.
2. Extract `DiscordRPCConfig`
3. PlayerSettingScreen still exists, because presents not-`PlayerConfig` settings - Identity and IdentityStorage.
4. Several fixes for `AutoConfig` 

notes: Language is part of SystemConfig, not PlayerConfig.

AutoConfig with PlayerConfig
![image](https://user-images.githubusercontent.com/2724433/112598331-7d2a8100-8e1f-11eb-8296-e4b699a63e44.png)


Player settings screen now.
![image](https://user-images.githubusercontent.com/2724433/112277160-85a18100-8c92-11eb-9542-c9f020174c79.png)


### How to test

1. Start terasology
2. Open `Setting`
3.a Open `autoconfig`
3.b Open `Player settings`
4. check screens at (3).

### Outstanding before merging

* [ ] __not related directly__ `SystemConfig`'s settings haven't names.
* [ ] __not related directly__ provide pretty screen/skin for AutoConfig Screen. ( design )
* [ ] Provide languages from translation subsystem to language constraint (idk how, translate subsystem initializing after config)   
     Fix Pirate language -  it exists in Widget, but absent at constraint.
* [X] Fix misalign settings in playerconfig
